src/MySTMarkdownCell.tsx

```diff
@@ -17,11 +17,28 @@ import { useParse } from 'myst-to-react';
 import { parseContent } from './myst';
 import { IMySTMarkdownCell } from './types';
 import { linkFactory } from './links';
+import { selectAll } from 'unist-util-select';
+
+import { PromiseDelegate } from '@lumino/coreutils';
+import { metadataSection, IUserExpressionMetadata } from './metadata';
+import { CellMetadataProvider } from './cellMetadataProvider';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
 
 export class MySTMarkdownCell
   extends MarkdownCell
   implements IMySTMarkdownCell
 {
+  private _doneRendering = new PromiseDelegate<void>();
+
+  private __rendermime: IRenderMimeRegistry;
+
+  constructor(options: MarkdownCell.IOptions) {
+    super(options);
+
+    this.__rendermime = options.rendermime.clone();
+    // this.__rendermime.addFactory(textRendererFactory);
```

Comment:
We should add or improve the inline rendering.

Reply:
@agoose77, looking more at the text renderer -- I think we might actually just want to do this in react even?

I think we are going to have to have a bunch of custom renderers, that are specific to the inline execution, but that can probably all come later!

Reply:
yes, so before defering to the standard rendermime renderers, take a different rendering route based on `mimetype` and skip the defaults altogether, for text etc...

For other things like widgets, we can maybe look at redefining/overriding the styles in CSS? we'd have to figure out how that plays with the `style` attributes on widgets themselves. https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Styling.html#The-style-attribute

Reply:
I *think* rendering plain-text in MyST might be reasonable. There's a big question over MyST integration in JLab as to how much we take over e.g. Markdown rendering currently goes through the default marked implementation.  

Perhaps we want to recognise a particular flavour of Markdown, so that users can opt-in to displaying rich text via outputs. Also, I wonder what myst-nb does here — how do `text/markdown` outputs get rendered (as MyST?)

Reply:
The downside of this is now other JupyterLab extensions can't override e.g. `text/plain` rendering, but it's not necessarily a problem; we just need to define what should be the behaviour here. My take is that we special-case `text/plain`, errors, LaTeX, and Markdown (maybe only if it's MyST markdown), and the rest goes through the rendermime interface.

Reply:
Nice, agree with all of that! Will pick it up in a future PR!